core/vm/interface.go

```diff
@@ -48,6 +48,10 @@ type StateDB interface {
 	// Arbitrum: preserve old empty account behavior
 	CreateZombieIfDeleted(common.Address)
 
+	// Arbitrum
```

Comment:
I'm surprised by this API. I would have expected that the api, would be something like, "Prevent transactions that have this (From and Nonce or some other unique transaction identifier) from being processed." But, this seems like it's just flipping some bit in the StateDB telling it to reject the "next" transaction or "next block." It would be great to have a few tests in this PR to demonstrate how callers can use this API to accomplish the desired filtering.

Reply:
added a nitro system_test in this [PR](https://github.com/OffchainLabs/nitro/pull/2807)